features/dual-stack-support.md

```diff
@@ -0,0 +1,229 @@
+
+[//]: # (User stories are a way to communicate user value. User stories follow the style)
+[//]: # (as a [type of user], I want [an action] so that [a benefit/a value]. Istio currently has no user)
+[//]: # (story template. Maybe you can make one?)
+
+[//]: # (User stories must be presented in a work group meeting. They need no approval and are later integrated)
+[//]: # (into the RFCs, which do need approval for alpha. You may find value to negotiate within the work group where the)
+[//]: # (user stories are presented to help clarify the user stories.)
+
+- [x] [RFC Authored] - [Supporting Dual Stack](https://docs.google.com/document/d/15LP2XHpQ71ODkjCVItGacPgzcn19fsVhyE7ruMGXDyU/edit?usp=sharing).
+
+[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
+[//]: # (meeting to graduate to experimental.)
+
+- [x] [Documentation](https://istio.io/latest/news/releases/1.17.x/announcing-1.17/#istio-dual-stack-support), as well as [Supporting Dual Stack](https://docs.google.com/document/d/15LP2XHpQ71ODkjCVItGacPgzcn19fsVhyE7ruMGXDyU/edit?usp=sharing) for enabling and using the feature.
+
+[//]: # (The documentation instructions may exist on the developer wiki or the team drive. They may include instructions)
+[//]: # (for building running a `istioctl experimental command`, or using the preview profile,)
+[//]: # (or any other relevant information.)
+
+- [x] [Feedback plan](insert_your_link_here). N/A, we already have feedback from users that this is desired and useful.
+
+[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)
+
+- [x] Disabled by default.
+
+- [x] No impact on performance when the feature is disabled.
+
+
+[//]: # (Once all other items are completed, features.yaml should be updated to promote the feature)
+
+- [x] [features.yaml](https://github.com/istio/enhancements/blob/master/features.yaml) updated for this feature
+---
+
+## Alpha
+
+### Requirements: 
+
+**Design**
+
+- [ ] RFC has been approved describing the intention of the feature as well as the user stories behind the feature. 
+N/A
+
+**Config**
+
+- [x] Explicit user action is required to enable this feature (Please refer to the `Installation` section in [Supporting Dual Stack](https://docs.google.com/document/d/15LP2XHpQ71ODkjCVItGacPgzcn19fsVhyE7ruMGXDyU/edit?usp=sharing) ). 
+
+> Link to instructions for enabling
+
+**Docs**
+
+- [x] Reference docs are published to preliminary.istio.io or the Istio wiki.
+- [ ] Basic feature docs are published on preliminary.istio.io describing what the feature does, how to use it, and any caveats. 
+- [x] Release notes entries added as appropriate
+- [ ] Upgrade notes entries added as appropriate
+
+**Tests**
+
+- [x] Automated integration tests cover core use cases with the feature enabled. 
```

Comment:
Do we actually have coverage of dual stack features? And not just "test suite runs on DS cluster"?

Reply:
From what I have seen the dual stack testing is not very comprehensive and some basic things like traffic through a gateway and some DNS cases that were broken weren't caught by any tests.   IMO we should still declare this an alpha feature,  people have been fiddling with IPv6 and dual stack for years,  but some catchup on test coverage would be a good idea if possible.

Reply:
Hi @howardjohn, for the integration tests, we have done following items:
1. Deployed dual stack k8s cluster, single stack Istio and dual stack services
2. Based on item 1 to test the connectivity and accessibility, the same functionality verification as IPv6 support.

Besides above, there also have some dual stack proxy unit test. If the connectivity and accessibility are the core use cases, I think yes, however, we also should deploy the Istio in dual stack mode, that's missed.

Reply:
I added this info in this part